### ISSUE TITLE
False positive `PossiblyUnusedMethod` on trait boot{Trait}/initialize{Trait} methods

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1069]
**Describe the bug**

Laravel boots traits by reflection: in `Illuminate\Database\Eloquent\Model::bootTraits()` it scans each used trait and dynamically calls any static method named `boot{TraitName}` (and, in `initializeTraits()`, `initialize{TraitName}`). These methods are never referenced explicitly anywhere in user code, so with `findUnusedCode` enabled Psalm reports them as `PossiblyUnusedMethod`:

```
ERROR: PossiblyUnusedMethod
Cannot find explicit calls to method App\Models\Article::bootHasUuid (but did find some potential callers) (see https://psalm.dev/087)
    final public static function bootHasUuid(): void
```

The plugin already understands a lot of Eloquent magic; it should also recognize these trait-boot/initialize hooks as used so they aren't flagged as dead code.

**Code to reproduce**

```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

trait HasUuid
{
    // Called by Laravel via reflection in Model::bootTraits(); never referenced explicitly.
    public static function bootHasUuid(): void
    {
        static::creating(static function (Model $model): void {
            $model->uuid = (string) Str::uuid();
        });
    }

    // Called by Laravel via reflection in Model::initializeTraits(); same problem.
    public function initializeHasUuid(): void
    {
        $this->mergeFillable(['uuid']);
    }
}

class Article extends Model
{
    use HasUuid;
}
```

Run Psalm with `findUnusedCode="true"` → both `bootHasUuid()` and `initializeHasUuid()` are reported as `PossiblyUnusedMethod`.

**Expected**

The plugin should suppress `PossiblyUnusedMethod`/`UnusedMethod` for methods matching `boot{TraitName}` and `initialize{TraitName}` when the declaring trait is used by an Eloquent model, since Laravel invokes them dynamically. (Same pattern Laravel itself relies on: `Model::bootTraits()` / `Model::initializeTraits()`.)

**Impacted Versions**

```text
laravel/framework      v12.62.0
psalm/plugin-laravel    dev-master
vimeo/psalm             7.0.0-beta19
```

**Psalm config**

Relevant bit is `findUnusedCode` enabled:

```xml
<psalm findUnusedCode="true" errorLevel="1">
    <plugins>
        <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
    </plugins>
</psalm>
```

**Additional context**

Reference for the dynamic dispatch: `Illuminate\Database\Eloquent\Model::bootTraits()` and `::initializeTraits()` build the method name from the trait's short class name (`'boot'.class_basename($trait)`) and call it via `method_exists()`/reflection. A possible fix is an `AfterClassLikeVisit`/method-usage hook that marks any trait method whose name is `boot` or `initialize` + the trait short name as used.
